src/overlays/actors/ovl_Obj_Skateblock/z_obj_skateblock.c

```diff
@@ -28,76 +39,655 @@ const ActorInit Obj_Skateblock_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A22308.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A22334.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A2244C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A224A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A22628.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A2264C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A22728.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A2273C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A227A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/func_80A227C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/ObjSkateblock_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Skateblock/ObjSkateblock_Draw.s")
+static s32 D_80A22A10 = 0;
+
+static s32 D_80A22A14 = 0;
+
+static TexturePtr D_80A22A18 = NULL;
+
+s32 func_80A21150(ObjSkateblock* this) {
+    s32 temp_v0;
+
+    if (fabsf(this->dyna.pushForce) > 0.1f) {
+        if (this->dyna.pushForce > 0.0f) {
+            temp_v0 = BINANG_ADD(this->dyna.yRotation, 0x2000);
+        } else {
+            temp_v0 = BINANG_SUB(this->dyna.yRotation, 0x6000);
+        }
+
+        if (temp_v0 < -0x4000) {
+            return 3;
+        }
+
+        if (temp_v0 < 0) {
+            return 1;
+        }
+
+        if (temp_v0 < 0x4000) {
+            return 2;
+        }
+
+        return 0;
+    }
+    return -1;
+}
+
+void func_80A211F4(ObjSkateblock* this, s32 arg1) {
+    s32 i;
+    s32 phi_v0 = false;
+
+    if (this->dyna.pushForce > 0.0f) {
+        phi_v0 = true;
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_172); i++) {
+        if ((arg1 == i) && phi_v0) {
+            this->unk_172[i]++;
+        } else if (this->unk_172[i] > 0) {
+            this->unk_172[i]--;
+        }
+    }
+}
+
+void func_80A212F0(Vec3f* arg0, Vec3f* arg1, s16 arg2) {
+    f32 sp1C = Math_SinS(arg2);
+    f32 temp_f0 = Math_CosS(arg2);
+
+    arg0->x = (arg1->z * sp1C) + (arg1->x * temp_f0);
+    arg0->y = arg1->y;
+    arg0->z = (arg1->z * temp_f0) - (arg1->x * sp1C);
+}
+
+s32 func_80A21370(ObjSkateblock* this, GlobalContext* globalCtx) {
+    static Vec2f D_80A22A1C[] = {
+        { -300.0f, 300.0f }, { 300.0f, 300.0f }, { -300.0f, -300.0f }, { 300.0f, -300.0f }, { 0.0f, 0.0f },
+    };
+    static Vec2f D_80A22A44[] = {
+        { 0.5f, -0.5f }, { -0.5f, -0.5f }, { 0.5f, 0.5f }, { -0.5f, 0.5f }, { 0.0f, 0.0f },
+    };
+    ObjSkateblockFloor* floor;
+    s32 i;
+    s32 pad;
+    s32 sp98;
+    Vec3f sp8C;
+    s32 sp88;
+    f32 phi_f20;
+
+    phi_f20 = BGCHECK_Y_MIN;
+    sp88 = false;
+    sp8C.y = this->dyna.actor.world.pos.y + 40.0f;
+    sp98 = -1;
```

Comment:
You should mix up those declarations and defintions

Reply:
I can't, I tried but I couldn't re-order them to fit the stack order.